packages/vite/src/node/build.ts

```diff
@@ -63,9 +62,9 @@ export interface BuildOptions {
    */
   target?: 'modules' | TransformOptions['target'] | false
   /**
-   * Whether to inject dynamic import polyfill. Defaults to `true`, unless
-   * `target` is `'esnext'`.
+   * whether to inject dynamic import polyfill.
    * Note: does not apply to library mode.
+   * @deprecated the dynamic import polyfill has been removed
```

Comment:
Let's add a reference to this PR in the deprecated message
```suggestion
   * @deprecated the dynamic import polyfill has been removed #2976
```
The rest LGTM!